### PR TITLE
Bc/email error

### LINKED ISSUE #23: Encountered error while restarting the server
It looks like there is a bug in the emailer, which might be why we are not receiving the reports from the server...

```
bachapin@mieserver:~/MIE-MCServer$ python main.py -r
Issued server command [say The server is being restarted in 60 seconds.]
TypeError('expected str, bytes or os.PathLike object, not BufferedReader')
Traceback (most recent call last):
  File "/home/bachapin/MIE-MCServer/main.py", line 359, in <module>
    main()
  File "/home/bachapin/MIE-MCServer/main.py", line 356, in main
    args.func(args)
  File "/home/bachapin/MIE-MCServer/main.py", line 104, in parse
    reboot.run()
  File "/home/bachapin/MIE-MCServer/scripts/reboot.py", line 13, in run
    email_log(subject, body)
  File "/home/bachapin/MIE-MCServer/util/syslog.py", line 58, in email_log
    mailer.send()
  File "/home/bachapin/MIE-MCServer/util/emailer.py", line 31, in send
    filename=basename(file))
  File "/usr/lib/python3.9/posixpath.py", line 142, in basename
    p = os.fspath(p)

Error in sys.excepthook:
Traceback (most recent call last):
  File "/home/bachapin/MIE-MCServer/util/emailer.py", line 31, in send
    filename=basename(file))
  File "/usr/lib/python3.9/posixpath.py", line 142, in basename
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not BufferedReader

Original exception was:
Traceback (most recent call last):
  File "/home/bachapin/MIE-MCServer/main.py", line 359, in <module>
    main()
  File "/home/bachapin/MIE-MCServer/main.py", line 356, in main
    args.func(args)
  File "/home/bachapin/MIE-MCServer/main.py", line 104, in parse
    reboot.run()
  File "/home/bachapin/MIE-MCServer/scripts/reboot.py", line 13, in run
    email_log(subject, body)
  File "/home/bachapin/MIE-MCServer/util/syslog.py", line 58, in email_log
    mailer.send()
  File "/home/bachapin/MIE-MCServer/util/emailer.py", line 31, in send
    filename=basename(file))
  File "/usr/lib/python3.9/posixpath.py", line 142, in basename
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not BufferedReader
```

### PR DESCRIPTION
**Pull request by BAChapin**
## GitHub Ticket

Resolves #23 

## What changed?
- Fixed the issue that was preventing it from restarting, which happened to be revolving around the Emailer failing.
- Also updated the email body for uncaught exceptions. It now reads better, is formatted better, and is using the proper signature. It is also issuing a call to action.

## Pull request type
Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): 
    - Updated a method of handling errors

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Your bug PR needs merged prior to this being merged into `v1.1.2`. I will need to rebase `v1.1.2` with `v1.1.1`. It probably wouldn't be a bad idea for me to rebase `bc/email-error` with those changes just to make sure there are not merge issues. Not that I expect there to be, but the CHANGELOG.md file might.

## Checklist
Please ensure the following items are taken care of as part of your PR. If they are not, your PR will NOT be approved!
- [x] You have added doc strings to any user-facing functions
- [x] You have updated the CHANGELOG.md file with your changes

### CODE REVIEW

File: util/emailer.py [hunk 1]
```diff
@@ -25,15 +25,16 @@ def send(self):
         msg.attach(MIMEText(self.body, "plain"))
 
         for attachment in self.attachments:
-            with open(attachment, "rb") as file:
-                aFile = MIMEApplication(file.read(), _subtype="txt")
-                aFile.add_header("Content-Disposition", "attachment",
-                    filename=basename(file))
-                msg.attach(file)
+            split = attachment.split('/')
+            name = split[-1]
```
Comment (Michaelcraun) [inline util/emailer.py:29]:
```
from os.path import basename, normpath
name = basename(normpath(attachment))
```

File: util/emailer.py [hunk 2]
```diff
@@ -25,15 +25,16 @@ def send(self):
...
+            file = MIMEApplication(open(attachment, 'rb').read(),
+                                   _subtype="txt")
+            file.add_header("Content-Disposition", "attachment", filename=name)
```
Comment (Michaelcraun) [inline util/emailer.py:32]: This file is never closed. Will that cause memory issues?

**Review by Michaelcraun — CHANGES_REQUESTED** [review]

File: util/emailer.py (hunk 2 above)
Comment (BAChapin) [reply util/emailer.py:32]: From my testing it doesn't seem to cause any memory leaks or anything, but I'll update this PR later today.

File: util/emailer.py (hunk 1 above)
Comment (BAChapin) [reply util/emailer.py]: Resolved!

File: util/emailer.py (hunk 2 above)
Comment (BAChapin) [reply util/emailer.py]: And Resolved!

Comment (Michaelcraun) [reply util/emailer.py]: Perfect!

**Review by Michaelcraun — APPROVED** [review]

**Comment by Michaelcraun** [comment]
Whenever you want to rebase, @BAChapin

**Comment by BAChapin** [comment]
@Michaelcraun Alright, its rebased. Good to merge.

**Pull request merged**